ner_v1/detectors/textual/text/text_detection_model.py

```diff
@@ -0,0 +1,93 @@
+                        u'Barnala': [u'', u'Barnala'],
+                        ...
+                        u'chennai': [u'', u'chennai', u'tamilnadu', u'madras'],
+                        u'hyderabad': [u'hyderabad'],
+                        u'koramangala': [u'koramangala']
+                    }
+
+            text_detection = TextModelDetector('city')
+            text_detection.detect_entity('Come to Chennai, TamilNadu,  I will visit Delhi next year')
+
+                Output:
+                    ([{'datastore_verified': True, 'value': u'Chennai'},
+                      {'datastore_verified': True, 'value': u'New Delhi'},
+                      {'datastore_verified': True, 'value': u'chennai'}]
+                    , ['chennai', 'delhi', 'tamilnadu'])
+
+            text_detection.tagged_text
+
+                Output:
+                    ' come to __city__, __city__,  i will visit __city__ next year '
+
+        Additionally this function assigns these lists to self.text_entity_values and self.original_texts attributes
+        respectively.
+        """
+        self._process_text(text)
+        values, original_texts = self._text_detection_with_variants()
+
+        text_entity_verified_values = TextModelDetector.add_verification_source(values=values,
+                                                                                verification_source_dict=
+                                                                                {DATASTORE_VERIFIED: True})
+        self.text_entity_values, self.original_texts = text_entity_verified_values, original_texts
+
+        return self.text_entity_values, self.original_texts
+
+    @staticmethod
+    def add_verification_source(values, verification_source_dict):
+        """
+        Add the verification source for the detected entities
+        Args:
+            values (list): List of detected text type entities
+            verification_source_dict (dict): Dict consisting of the verification source and value.
+        Returns:
+            text_entity_verified_values (list): List of dicts consisting of the key and values for the keys
+            value and verification source
+        Example:
+            values = [u'Chennai', u'New Delhi', u'chennai']
+            verification_source_dict = {"datastore_verified": True}
+
+            >> add_verification_source(values, verification_source_dict)
+                [{'datastore_verified': True, 'value': u'Chennai'},
+                 {'datastore_verified': True, 'value': u'New Delhi'},
+                 {'datastore_verified': True, 'value': u'chennai'}]
+        """
+        text_entity_verified_values = []
+        for text_entity_value in values:
+            text_entity_dict = {ENTITY_VALUE_DICT_KEY: text_entity_value}
+            for verification_source_key in list(verification_source_dict.keys()):
+                text_entity_dict[verification_source_key] = verification_source_dict[verification_source_key]
```

Comment:
Line 90-91 can be simplified as 
text_entity_dict.update(verification_source_dict)